packages/teleterm/src/ui/DocumentTerminal/useTshFileTransferHandlers.ts

```diff
@@ -47,14 +46,14 @@ function transferFile(
   appContext: IAppContext,
   originatingDocumentUri: string,
   file: FileTransferRequestObject,
-  fileTransferListeners: FileTransferListeners,
   abortController: AbortController,
   direction: FileTransferDirection
-): void {
+): FileTransferListeners {
   const server = appContext.clustersService.getServer(file.serverUri);
+  const eventsEmitter = createFileTransferEventsEmitter();
```

Comment:
Yes, `appContext.fileTransferClient.transferFile` already returns an object that we can use, but I need the event emitter to create a layer between `tshClient` and `FileTransfer`. Events returned from the `tshClient` are used inside `useTshFileTransferHandlers` and `FileTransfer` uses these returned from `useTshFileTransferHandlers`.
It is needed, because the errors needs to go through `retryWithRelogin` which decides what to do with them first.
For example, when there is `ssh: cert has expired` error should show to the user only when he reject to log in. 

I wanted to avoid this, but I don't see any other way not to create an event emitter.